In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('./sales_data_with_discounts.csv')
df

Date       Day  SKU City  Volume         BU  Brand       Model  \
0    01-04-2021  Thursday  M01    C      15    Mobiles  RealU       RU-10   
1    01-04-2021  Thursday  M02    C      10    Mobiles  RealU   RU-9 Plus   
2    01-04-2021  Thursday  M03    C       7    Mobiles   YouM       YM-99   
3    01-04-2021  Thursday  M04    C       6    Mobiles   YouM  YM-99 Plus   
4    01-04-2021  Thursday  M05    C       3    Mobiles   YouM      YM-98    
..          ...       ...  ...  ...     ...        ...    ...         ...   
445  15-04-2021  Thursday  L06    C       2  Lifestyle  Jeera   M-Casuals   
446  15-04-2021  Thursday  L07    C       6  Lifestyle   Viva   W-Western   
447  15-04-2021  Thursday  L08    C       2  Lifestyle   Viva    W-Lounge   
448  15-04-2021  Thursday  L09    C       3  Lifestyle  Jeera   M-Formals   
449  15-04-2021  Thursday  L10    C       1  Lifestyle  Jeera     M-Shoes   

     Avg Price  Total Sales Value  Discount Rate (%)  Discount Amount  \
0        12100             181500          11.654820     21153.498820   
1        10100             101000          11.560498     11676.102961   
2        16100             112700           9.456886     10657.910157   
3        20100             120600           6.935385      8364.074702   
4         8100              24300          17.995663      4372.946230   
..         ...                ...                ...              ...   
445       1300               2600          15.475687       402.367873   
446       2600              15600          17.057027      2660.896242   
447       1600               3200          18.965550       606.897606   
448       1900               5700          16.793014       957.201826   
449       3100               3100          15.333300       475.332295   

     Net Sales Value  
0      160346.501180  
1       89323.897039  
2      102042.089843  
3      112235.925298  
4       19927.053770  
..               ...  
445      2197.632127  
446     12939.103758  
447      2593.102394  
448      4742.798174  
449      2624.667705  

[450 rows x 13 columns]

#### Descriptive Analytics for Numerical Columns

In [12]:
df.dtypes

Date                  object
Day                   object
SKU                   object
City                  object
Volume                 int64
BU                    object
Brand                 object
Model                 object
Avg Price              int64
Total Sales Value      int64
Discount Rate (%)    float64
Discount Amount      float64
Net Sales Value      float64
dtype: object

In [17]:
numcol = df[['Volume', 'Avg Price', 'Total Sales Value', 'Discount Rate (%)', 'Discount Amount', 'Net Sales Value']]
numcol.head()

Volume  Avg Price  Total Sales Value  Discount Rate (%)  Discount Amount  \
0        15      12100             181500          11.654820     21153.498820   
1        10      10100             101000          11.560498     11676.102961   
2         7      16100             112700           9.456886     10657.910157   
3         6      20100             120600           6.935385      8364.074702   
4         3       8100              24300          17.995663      4372.946230   
..      ...        ...                ...                ...              ...   
445       2       1300               2600          15.475687       402.367873   
446       6       2600              15600          17.057027      2660.896242   
447       2       1600               3200          18.965550       606.897606   
448       3       1900               5700          16.793014       957.201826   
449       1       3100               3100          15.333300       475.332295   

     Net Sales Value  
0      160346.501180  
1       89323.897039  
2      102042.089843  
3      112235.925298  
4       19927.053770  
..               ...  
445      2197.632127  
446     12939.103758  
447      2593.102394  
448      4742.798174  
449      2624.667705  

[450 rows x 6 columns]

In [19]:
numcol.describe()

Volume     Avg Price  Total Sales Value  Discount Rate (%)  \
count  450.000000    450.000000         450.000000         450.000000   
mean     5.066667  10453.433333       33812.835556          15.155242   
std      4.231602  18079.904840       50535.074173           4.220602   
min      1.000000    290.000000         400.000000           5.007822   
25%      3.000000    465.000000        2700.000000          13.965063   
50%      4.000000   1450.000000        5700.000000          16.577766   
75%      6.000000  10100.000000       53200.000000          18.114718   
max     31.000000  60100.000000      196400.000000          19.992407   

       Discount Amount  Net Sales Value  
count       450.000000       450.000000  
mean       3346.499424     30466.336131  
std        4509.902963     46358.656624  
min          69.177942       326.974801  
25%         460.459304      2202.208645  
50%         988.933733      4677.788059  
75%        5316.495427     47847.912852  
max       25738.022194    179507.479049

In [23]:
import numpy as np
from scipy import stats

In [50]:
dicts = []
for i in numcol.columns:
    dicts.append({
    'mean' : round(np.mean(numcol[i]), 2),
    'median' : round(np.median(numcol[i]), 2),
    'mode' : stats.mode(numcol[i]).mode,
    'std' : round(np.std(numcol[i]), 2)
})

In [52]:
result = pd.DataFrame({
    'Volume' : dicts[0],
    'Avg Price' : dicts[1],
    'Total Sales Value' : dicts[2],
    'Discount Rate (%)' : dicts[3],
    'Discount Amount' : dicts[4],
    'Net Sales Value' : dicts[5]
})

result

Volume  Avg Price  Total Sales Value  Discount Rate (%)  \
mean      5.07   10453.43           33812.84          15.160000   
median    4.00    1450.00            5700.00          16.580000   
mode      3.00     400.00           24300.00           5.007822   
std       4.23   18059.80           50478.89           4.220000   

        Discount Amount  Net Sales Value  
mean        3346.500000     30466.340000  
median       988.930000      4677.790000  
mode          69.177942       326.974801  
std         4504.890000     46307.120000

## Interpretations

#### Volume:
* Mean Volumn sold in 5.07 units per transaction.
* The most frequently occuring volume in a single transaction is 3 units.

#### Avg Price:
* Average Price per transaction is 10,453.43, with a median price of 1,450.00.
* The most common average price is 400.

#### Total Sales Value
* Average Total Sale Value is 33,812.84, with a median of 5,700.00.
* The most Frequent Sale Value is 24,300.00.

#### Discount Rate (%)
* Average Discount Applied is 15.16%, with a median discount value of 16.58%.
* The Most Common Discount Rate is 5.01%.

#### Discount Amount
* Average Discount Amount per Transaction is 3,346.5, with a median discount of 988.93.
* The Most frequent discount is of 69.18.

#### Net Sales Value
* Average Sale Value after Discount is 30,466.34, with a median value of 4,677.79.
* The Most common net sale value is approximately 326.97.